python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -178,29 +202,49 @@ def asset_daemon_cursor_to_instigator_serialized_cursor(cursor: AssetDaemonCurso
     return VERSION + encoded_cursor
 
 
-def asset_daemon_cursor_from_instigator_serialized_cursor(
-    serialized_cursor: Optional[str], asset_graph: Optional[AssetGraph]
+def asset_daemon_cursor_from_serialized_cursors(
+    serialized_sensor_cursor: Optional[str],
+    asset_graph: Optional[AssetGraph],
+    serialized_asset_cursors: Optional[Mapping[AssetKey, str]] = None,
 ) -> AssetDaemonCursor:
     """This method decompresses the serialized cursor and returns a deserialized cursor object,
     converting from the legacy cursor format if necessary.
     """
-    if serialized_cursor is None:
+    if serialized_sensor_cursor is None:
         return AssetDaemonCursor.empty()
 
-    version, encoded_bytes = serialized_cursor[0], serialized_cursor[1:]
+    version, encoded_bytes = serialized_sensor_cursor[0], serialized_sensor_cursor[1:]
     if version != "0":
         return AssetDaemonCursor.empty()
 
     decoded_bytes = base64.b64decode(encoded_bytes)
     decompressed_bytes = zlib.decompress(decoded_bytes)
     decompressed_str = decompressed_bytes.decode("utf-8")
 
-    deserialized_cursor = deserialize_value(
+    raw_sensor_cursor = deserialize_value(
         decompressed_str, (LegacyAssetDaemonCursorWrapper, AssetDaemonCursor)
     )
-    if isinstance(deserialized_cursor, LegacyAssetDaemonCursorWrapper):
-        return deserialized_cursor.get_asset_daemon_cursor(asset_graph)
-    return deserialized_cursor
+
+    sensor_cursor = (
+        raw_sensor_cursor.get_asset_daemon_cursor(asset_graph)
+        if isinstance(raw_sensor_cursor, LegacyAssetDaemonCursorWrapper)
+        else raw_sensor_cursor
+    )
+    asset_eval_states = {
+        k: deserialize_value(
+            serialized_cursor, AssetScopedAssetDaemonCursor
+        ).previous_evaluation_state
+        for k, serialized_cursor in (serialized_asset_cursors or {}).items()
+    }
+
+    states_by_key = {
+        **sensor_cursor.previous_evaluation_state_by_key,
+        **asset_eval_states,
+    }
+
+    return dataclasses.replace(
+        sensor_cursor, previous_evaluation_state=list(states_by_key.values())
```

Comment:
despite being a list this does not have a consistnet ordering right? guess that ship has sailed / is already true of the existing AssetDaemonCursor

Reply:
correct